consistent-sampling/src/main/java/io/opentelemetry/contrib/samplers/ConsistentSampler.java

```diff
@@ -212,8 +240,7 @@ public final SamplingResult shouldSample(
 
     // generate new r-value if not available
     if (!otelTraceState.hasValidR()) {
-      otelTraceState.setR(
-          Math.min(randomGenerator.numberOfLeadingZerosOfRandomLong(), OtelTraceState.getMaxR()));
+      otelTraceState.setR(Math.min(rValueGenerator.applyAsInt(traceId), OtelTraceState.getMaxR()));
```

Comment:
I'd love to see this approach made more general with rValueGenerator taking not only traceId, but also the Attributes as arguments.
The use case for this is to assign the same rValue to a group of traces, so they are likely to survive sampling as an intact group - still while ensuring the correct statistical distribution of the rValues over the whole trace population.